app/models/program_session.rb

```diff
@@ -15,6 +15,26 @@ class ProgramSession < ActiveRecord::Base
   scope :unscheduled, -> do
     where(state: ACTIVE).where.not(id: TimeSlot.pluck(:program_session_id))
   end
+
+  def self.create_from_proposal(proposal)
+    self.transaction do
+      #create program session object
+      ps = ProgramSession.create!(event_id: proposal.event_id,
+                                  proposal_id: proposal.id,
+                                  title: proposal.title,
+                                  abstract: proposal.abstract,
+                                  track_id: proposal.track_id,
+                                  session_format_id: proposal.session_format_id)
+
+      #attach proposal speakers to new program session
+      proposal.speakers.each do |speaker|
+        speaker.program_session_id = ps.id
+        speaker.speaker_name = speaker.user.name
+        speaker.speaker_email = speaker.user.email
```

Comment:
Make sure not to overwrite speaker_name or speaker_email if they already have values. If they've already been set, it means that someone has already taken some time to customize those values for that speaker and we don't want to undo that.